hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -462,6 +476,20 @@ public Response head(
     OzoneKey key;
     try {
       key = getClientProtocol().headS3Object(bucketName, keyPath);
+
+      /*
+        Necessary for directories in buckets with FSO layout.
+        Intended for apps which use Hadoop S3A.
+        Example of such app is Trino (through Hive connector).
+       */
+      boolean isFsoDirCreationEnabled = ozoneConfiguration
```

Comment:
nit: to avoid duplication, this can be extracted as a function with `isFile` and `keyPath` as parameters.